evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -579,6 +579,33 @@ def test_make_component_list_from_actions():
                                                           TargetImputer(impute_strategy="most_frequent")]
 
 
+@pytest.mark.parametrize("samplers", [None, "Undersampler", "SMOTE Oversampler", "SMOTENC Oversampler", "SMOTEN Oversampler"])
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass', 'regression'])
+def test_make_pipeline_samplers(problem_type, samplers, X_y_binary, X_y_multi, X_y_regression, has_minimal_dependencies):
+    if problem_type == 'binary':
+        X, y = X_y_binary
+    elif problem_type == 'multiclass':
+        X, y = X_y_multi
+    else:
+        X, y = X_y_regression
+    estimators = get_estimators(problem_type=problem_type)
+
+    for estimator in estimators:
+        pipeline = make_pipeline(X, y, estimator, problem_type, sampler_name=samplers)
+        if has_minimal_dependencies and samplers is not None:
+            samplers = 'Undersampler'
+        # check that we do add the sampler properly
+        component_to_check = pipeline.component_graph
+        if samplers is not None and problem_type != 'regression':
+            # we add the sampler before the scaler if it exists
+            if component_to_check[-2].name == 'Standard Scaler':
+                assert component_to_check[-3].name == samplers
+            else:
+                assert component_to_check[-2].name == samplers
```

Comment:
Nit-pick: just reference `pipeline.component_graph` directly here instead of declaring variable `component_to_check`, fewer mental hops to make.